### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
This will simplify the existing code and allow to make new changes easier.

### CODE REVIEW

**Comment by SemenchenkoVitaliy** [comment]
ping @lundibundi

File: lib/generator.js [hunk 1]
```diff
@@ -40,24 +40,29 @@ const links = [
   ['this', baseUrl + 'Web/JavaScript/Reference/Operators/this'],
 ];
 
-const getLinks = (usedTypes, customLinks = []) => {
+const getLinks = (usedTypes, customLinks) => {
   const result = common
     .merge(customLinks, links)
     .filter(([type]) => usedTypes.has(type))
     .map(([type, link]) => '[' + type.toLowerCase() + ']: ' + link);
   return result;
```
Comment (lundibundi) [inline lib/generator.js]: Inline result?

File: lib/generator.js [hunk 2]
```diff
@@ -188,112 +294,108 @@ const generateParameters = (parameters, customTypes, usedTypes) => {
...
-      return false;
-    }
-  }
-  return true;
-};
+const generateFunction = (name, sig, opts, methods) => {
+  if (sig.private && !opts.private) return [];
+  if (sig.instance) return generateInstance(name, sig, opts);
+
+  const buf = [
```
Comment (lundibundi) [inline lib/generator.js]: Also, inline?

**Comment by lundibundi** [comment]
Landed in a0c50391dd761973799d4e75f9b4e5a437da2c46.

**Pull request closed without merging**